DataAccessLibrary/Repositories/Generic/Repository.cs

```diff
@@ -1,61 +1,64 @@
 ﻿using System;
 using System.Collections.Generic;
-using System.Data.SqlClient;
 using System.Linq;
-using System.Reflection;
 using System.Text;
 using System.Threading.Tasks;
-using Dapper;
-using DataAccessLibrary.Repositories.Generic;
-using Microsoft.Extensions.Configuration;
-using DataAccessLibrary;
+using DataAccessLibrary.Base;
 
 
 namespace DataAccessLibrary.Repositories.Generic
 {
     // TODO change T to eg. IDataTable - with eg. property TableName
     // so we can know more about the class
-    public class Repository<T> : IRepository<T> where T : class, new()
+    public class Repository<T> : IRepository<T>
+        where T : Domain
     {
         private readonly ISqlDataAccess _database;
+
         private readonly string _tableName;
+        
         private readonly List<string> _properties;
 
         public Repository(ISqlDataAccess database)
         {
-            _database = database;
+            _database = database ?? throw new ArgumentNullException(nameof(database));
             _tableName = typeof(T).GetField("TableName")?.GetValue(null)?.ToString();
             _properties = typeof(T).GetProperties().Select(p => p.Name).ToList();
         }
 
         // TODO add error catching (eg. what if type do not have table in repository)
-        public  Task<List<T>> GetAll()
+        public  Task<List<T>> GetAllAsync()
         {
             var query = $"Select * From {_tableName};";
+
             return _database.LoadData<T, object>(query, new {});
         }
 
-        public Task<T> GetById(Guid id)
+        public Task<T> GetByIdAsync(Guid id)
         {
             var query = $"Select * From {_tableName} Where [Id] = @Id;";
+
             return _database.LoadData<T, object>(query, new {Id = id })
                 .ContinueWith(t => t.Result.FirstOrDefault());
         }
-        public Task DeleteById(Guid id)
+        public Task DeleteByIdAsync(Guid id)
         {
             var query = $"Delete From {_tableName} Where [Id] = @Id;";
+
             return _database.SaveData(query, new { Id = id });
         }
 
-        public Task Insert(T entity)
+        public Task InsertAsync(T entity)
         {
             var query = CreateInsertQuery();
+
             return _database.SaveData(query, entity);
         }
 
-        public Task Update(T entity)
+        public Task UpdateAsync(T entity)
```

Comment:
To samo jak wyżej, jeśli `T Entity` bierzesz bezpośrednio z inputu haker może próbować samemu ustawiać sobie pola których ty nie przewidziałaś.